R/modified_update.R

```diff
@@ -532,3 +537,98 @@ mu_calculate_rate <- function(name, dat, window, n_years) {
 
   dat
 }
+
+mu_fix_sdf7_psu <- function(meta, data){
+  ### this function fixes SDF7 - PSU - Measles - Campaign impacts
+  i <- meta$group$touchstone_src == "201210gavi-201303gavi-1" & 
+    meta$group$modelling_group == "PSU-Ferrari" &
+    meta$group$vaccine == "Measles"
+  dat_id <- meta$group$index[i]
+  stopifnot(length(dat_id) == 2)
+  
+  ## keep the rest
+  keep <- data[!(data$index %in% dat_id), ]
+  ## identify problem source
+  dat1 <- data[data$index == dat_id[1], ]
+  dat2 <- data[data$index == dat_id[2], ]
+  ## operation
+  dat <- dat1
+  i <- is.na(dat1$deaths_averted) | dat1$deaths_averted == 0.
+  
+  dat$deaths_averted[i] <- dat2$deaths_averted[i]
+  dat$cases_averted[i] <- dat2$cases_averted[i]
+  dat$fvps[i] <- dat2$fvps[i]
+  dat$coverage_old[i] <- dat2$coverage_old[i]
+  dat$coverage_target[i] <- dat2$coverage_target[i]
+  
+  
+  i <- sum(dat$deaths_averted, na.rm = TRUE) - 
+    (sum(dat1$deaths_averted, na.rm = TRUE) + sum(dat2$deaths_averted, na.rm = TRUE))
+  stopifnot( abs(i) < 1 )
+  
+  ### re-calculate impact rates
+  i <- is_blank(dat$fvps) &
+    !(is_blank(dat$deaths_averted) & is_blank(dat$cases_averted)) &
+    !is_blank(dat$coverage_target)
+  if (any(i)) {
+    stopifnot(x$activity_type == "campaign")
+    message("Creating synthetic fvps for ", x$model)
+    ## Otherwise use pop_routine.
+    dat$fvps[i] <- dat$coverage_old[i] * dat$coverage_target[i]
+  }
+  
+  i <- is_blank(dat$fvps) & !is_blank(dat$coverage_old)
+  if (any(i)) {
+    message("Creating synthetic fvps for ", x$model)
+    if((x$activity_type == "campaign")){
+      dat$fvps[i] <- dat$coverage_old[i] * dat$coverage_target[i]
+    }else{
+      dat$fvps[i] <- dat$coverage_old[i] * dat$pop_routine[i]
+    }
+  }
+  
+  year_min <- 2001
```

Comment:
should these values be passed in from the calling function?  I'm not really sure what this function is actually doing

Reply:
I will add some more comments to this function.